lib/flipper/adapters/memcached.rb

```diff
@@ -0,0 +1,74 @@
+require 'dalli'
+
+module Flipper
+  module Adapters
+    class Memcached
+      include Flipper::Adapter
+      FeaturesKey =  :flipper_features
+
+      attr_accessor :cache
+      attr_accessor :name
+
+      def initialize(adapter, address = 'localhost:11211', options = {})
+        @adapter = adapter
+        @name = adapter.name
+        @cache = Dalli::Client.new(address, options)
+      end
+
+      def features
+        cached(FeaturesKey) do
+          @adapter.features
+        end
+      end
+
+      def add(feature)
+        @adapter.add(feature).tap do
+          @cache.delete(FeaturesKey)
+        end
+      end
+
+      def remove(feature)
+        @adapter.remove(feature).tap do
+          @cache.delete(FeaturesKey)
+        end
+      end
+
+      def clear(feature)
+        @adapter.clear(feature).tap do
+          @cache.delete(feature)
+        end
+      end
+
+      def flush
+        @cache.flush
+      end
+
+      def get(feature)
+        cached(feature) do
+          @adapter.get(feature)
+        end
+      end
+
+      def enable(feature, gate, thing)
+        @adapter.enable(feature, gate, thing).tap do
+          @cache.delete(feature)
+        end
+      end
+
+      def disable(feature, gate, thing)
+        @adapter.disable(feature, gate, thing).tap do
+          @cache.delete(feature)
+        end
+      end
+
+      private
+
+      def cached(value)
```

Comment:
I'd drop this method and use Client#fetch in dalli directly in the methods above. get/set can be racey as last write wins whereas get/add is not.